aten/src/ATen/native/cuda/Normalization.cuh

```diff
@@ -1706,29 +1706,29 @@ at::Tensor batch_norm_backward_elemt_channels_last_cuda_template(
     if (weight.defined() && weight.scalar_type() != input.scalar_type()) {
       batch_norm_backward_elemt_channels_last_kernel<ELEMENTS_PER_ITER>
           <<<grid, block, 0, stream>>>(
-          grad_output.data_ptr<scalar_t>(),
-          input.data_ptr<scalar_t>(),
-          mean.data_ptr<accscalar_t>(),
-          inv_std.data_ptr<accscalar_t>(),
-          weight.data_ptr<accscalar_t>(),
-          sum_dy.data_ptr<accscalar_t>(),
-          sum_dy_xmu.data_ptr<accscalar_t>(),
-          grad_input.data_ptr<scalar_t>(),
+          grad_output.const_data_ptr<scalar_t>(),
+          input.const_data_ptr<scalar_t>(),
+          mean.const_data_ptr<accscalar_t>(),
+          inv_std.const_data_ptr<accscalar_t>(),
+          weight.const_data_ptr<accscalar_t>(),
+          sum_dy.const_data_ptr<accscalar_t>(),
+          sum_dy_xmu.const_data_ptr<accscalar_t>(),
+          grad_input.mutable_data_ptr<scalar_t>(),
           static_cast<accscalar_t>(norm_fct),
           reduction_size,
           stride);
           C10_CUDA_KERNEL_LAUNCH_CHECK();
     } else {
       batch_norm_backward_elemt_channels_last_kernel<ELEMENTS_PER_ITER>
           <<<grid, block, 0, stream>>>(
-          grad_output.data_ptr<scalar_t>(),
-          input.data_ptr<scalar_t>(),
-          mean.data_ptr<accscalar_t>(),
-          inv_std.data_ptr<accscalar_t>(),
-          weight.defined() ? weight.data_ptr<scalar_t>() : nullptr,
-          sum_dy.data_ptr<accscalar_t>(),
-          sum_dy_xmu.data_ptr<accscalar_t>(),
-          grad_input.data_ptr<scalar_t>(),
+          grad_output.const_data_ptr<scalar_t>(),
+          input.const_data_ptr<scalar_t>(),
+          mean.const_data_ptr<accscalar_t>(),
+          inv_std.const_data_ptr<accscalar_t>(),
+          weight.defined() ? weight.const_data_ptr<scalar_t>() : nullptr,
+          sum_dy.const_data_ptr<accscalar_t>(),
+          sum_dy_xmu.const_data_ptr<accscalar_t>(),
```

Comment:
You don't consistently const/mut sum_dy and sum_dy_xmu in this file. which is correct

Reply:
Which is _not_ correct, I'm assuming you meant?

It's not caught by CI, and is not caught by our build, which is why I missed it. Does this mean it's dead code?

Reply:
Is it mutable or const?

In general the mutable annotations must be audited way more closely, because int* will always silently decay to const int* and so it's easy to say something mutates when it doesn't

Reply:
Agreed, mutable must be audited more closely. Arguably const don't need to be audited at all, since the compiler does that for us.

sum_dy and sumn_dy are different, that's what you're noticing.
sum_dy_xmu is mutable after it's declared with empty and const where it's not. This is concordant with the functions that are called:
 * batch_norm_backward_elemt_channels_last_kernel const here
 * batch_norm_backward_reduce_channels_last_kernel mutable here